PracticeGit/test.c

```diff
@@ -0,0 +1,125 @@
+// -------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// -------------------------------------------------------------------------------------------------
+// test.c - Function that tests the program and main.
+// ------------------------------------------------------------------------------------------------
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <limits.h>
+#include "Program.h"
+
+#define RED     "\033[1;31m"
+#define GREEN   "\033[1;32m"
+#define CYAN    "\033[1;36m"
+#define RESET   "\033[0m"
+
+/// <summary>Execute palindrome test cases </summary>
+void RunPalindromeTest ();
+
+/// <summary>Execute Integer Reversal test cases </summary>
+void RunIntegerReversalTest ();
+
+///  <summary>Get user input for Palindrome checker </summary>
+void GetPalindromeInput ();
+
+///  <summary>Get user input for Integer reversal </summary>
+void GetIntegerInput ();
+
+void RunPalindromeTest () {
+   const char* testStrings[] = { "Eva, can I stab bats in a cave?", "malaya&lam", "Hey!! World", "J", "No lemon, no melon","$$$",
+                                 "Her favorite number is 2", "    ra   c eca  r " };
+   int expectedResults[] = { PALINDROME, PALINDROME, NOT_PALINDROME, PALINDROME, PALINDROME,NOT_PALINDROME, NOT_PALINDROME, PALINDROME };
```

Comment:
We try to limit the max number of chars in a single line to 100 wherever possible for readability purposes. You can find the char count in the bottom right corner of your text editor in VS as "Ch: _"
```suggestion
const char* testStrings[] = { "Eva, can I stab bats in a cave?", "malaya&lam", "Hey!! World",
                              "J", "No lemon, no melon","$$$", "Her favorite number is 2",
                              "    ra   c eca  r " };
   int expectedResults[] = { PALINDROME, PALINDROME, NOT_PALINDROME, PALINDROME, PALINDROME,NOT_PALINDROME, NOT_PALINDROME, PALINDROME };
```